griptape/drivers/video_generation/dream_machine_video_generation_driver.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+import logging
+import time
+from typing import TYPE_CHECKING, Any
+
+import requests
+from attrs import Factory, define, field
+
+from griptape.artifacts import VideoArtifact
+from griptape.drivers import BaseVideoGenerationDriver
+from griptape.utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from lumaai import LumaAI
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class DreamMachineVideoGenerationDriver(BaseVideoGenerationDriver):
+    api_key: str = field(kw_only=True, metadata={"serializable": True})
+    client: LumaAI = field(
+        default=Factory(
+            lambda self: import_optional_dependency("lumaai").LumaAI(auth_token=self.api_key), takes_self=True
+        ),
+        kw_only=True,
+    )
+    params: dict[str, Any] = field(default={}, kw_only=True, metadata={"serializable": True})
+
+    def try_text_to_video(self, prompt: str) -> VideoArtifact:
+        response = self.client.generations.create(prompt=prompt, **self.params)
+        generation = response
+        status = generation.state
+        while status in ["dreaming", "queued"]:
+            time.sleep(5)
+            if not generation.id:
+                raise Exception("Generation ID not found in the response")
+
+            generation = self.client.generations.get(generation.id)
+            status = generation.state
+        if status == "completed":
+            video_url = generation.assets.video
+            if not video_url:
+                raise Exception("Video URL not found in the generation response")
+            video_binary = self._download_video(video_url)
+            return VideoArtifact(
+                value=video_binary,
+            )
+        else:
+            raise Exception(f"Video generation failed with status: {status}")
+
+    def _download_video(self, video_url: str) -> bytes:
```

Comment:
videos are big and take a long time to come down. Do we...
1. ...already have this functionality somewhere else in the framework?
2. ...need to take into consideration fails on memory, disk, retries, timeout?

Reply:
id be fine with putting this functionality directly on `VideoArtifact` as a "lazy load" type feature. the alternative is creating a persistence driver for everything, which we dont have, but we have the `ArtifactFileOutputMixin` available.

Reply:
Agreed that the Driver should probably not be the one to do this. I don't know about putting it in `VideoArtifact` either -- feels like too much responsibility.

In-fact, we probably shouldn't make any assumptions that the user wants us to download it. Maybe they're fine receiving a URL that they watch in their browser. Maybe this should fall onto a Loader?

Reply:
could be the same idea as a lazy load. `VideoArtifact` contains a reference to the video somewhere.